tests/integration/test_bundle.py

```diff
@@ -0,0 +1,46 @@
+import subprocess
+
+import pytest
+import requests
+from pytest_operator.plugin import OpsTest
+from tenacity import retry, stop_after_delay, wait_fixed
+
+BUNDLE_PATH = "./releases/latest/edge/mlflow/bundle.yaml"
+MLFLOW_APP_NAME = "mlflow-server"
+
+
+@pytest.fixture
+def forward_connections():
+    mlflow_process = subprocess.Popen(
+        ["kubectl", "-n", "kubeflow", "port-forward", "pod/mlflow-server-0", "5002:5000"]
+    )
+
+    exporter_process = subprocess.Popen(
+        ["kubectl", "-n", "kubeflow", "port-forward", "pod/mlflow-server-0", "8002:8000"]
+    )
+    yield
+    mlflow_process.terminate()
+    exporter_process.terminate()
+
+
+class TestCharm:
+    @pytest.mark.abort_on_fail
+    async def test_deploy_bundle_works(self, ops_test: OpsTest):
+        subprocess.Popen(["juju", "deploy", f"{BUNDLE_PATH}", "--trust"])
+        ops_test.deploy_bundle(BUNDLE_PATH)
```

Comment:
should both of these be here?

Reply:
SO I am doing retry for the later test If I connect them the retry will try do redeploy the bundle every time